src/main/java/org/dependencytrack/event/kafka/processor/VulnerabilityScanResultProcessor.java

```diff
@@ -904,6 +927,24 @@ ON CONFLICT ("VULNERABILITY_ID", "COMPONENT_ID", "PROJECT_ID") DO UPDATE
                 """)
         void createAnalysisComments(@BindMethods final Iterable<AnalysisComment> comment);
 
+        @SqlQuery("""
+                UPDATE "ANALYSIS"
+                    SET "SUPPRESSED" = true, "STATE" = 'NOT_AFFECTED'
+                    WHERE "COMPONENT_ID" = :componentId
+                    AND "SUPPRESSED" = 'false'
+                    AND "VULNERABILITY_ID" IN
+                        ( SELECT "VULNERABILITY"."ID"
+                            FROM "VULNERABILITY"
+                            WHERE "VULNERABILITY"."SOURCE" = 'SNYK'
+                            <#if vulnIdsToExclude>
+                                AND "VULNERABILITY"."VULNID" != ANY(:vulnIdsToExclude)
+                            </#if>
+                        )
```

Comment:
Consider expressing this as a join instead. `UPDATE` has the `FROM` clause for that purpose. For example:

```sql
UPDATE "ANALYSIS"
   SET "SUPPRESSED" = TRUE
     , "STATE" = 'NOT_AFFECTED'
  FROM "VULNERABILITY"
 WHERE "ANALYSIS"."COMPONENT_ID" = :componentId
   AND "ANALYSIS"."VULNERABILITY_ID" = "VULNERABILITY"."ID"
   AND "VULNERABILITY"."SOURCE" = 'SNYK'
<#if vulnIdsToExclude>
   AND "VULNERABILITY"."VULNID" != ANY(:vulnIdsToExclude)
</#if>
```

If `vulnIdsToExclude` would be `null`, this subquery:

```sql
SELECT "VULNERABILITY"."ID"
  FROM "VULNERABILITY"
 WHERE "VULNERABILITY"."SOURCE" = 'SNYK'
```

would yield a rather large result set, which you can avoid with the join as shown above.